### PR TITLE
feat(push-notifications): add unregister functions

### PR DESCRIPTION
**Pull request by markemer**
Closes: #1100, Related to PR #1110 which this supersedes. We can close that PR when this merges.

### CODE REVIEW

**Comment by markemer** [comment]
> The new method needs to be added to the plugin interface in the definitions.ts file https://github.com/ionic-team/capacitor-plugins/blob/main/push-notifications/src/definitions.ts#L31
> 
> And also run `npm run build` so the method appears in the plugin docs
> 
> And would be good if the test app gets updated with a new button on the push notifications page to call the unregister method

Duh, yeah - I knew a forgot something - I was so preoccupied with thinking I forgot something on the native side, I forgot all the typescript.

**Review by giralte-ionic — APPROVED** [review]

File: push-notifications/ios/Plugin/PushNotificationsPlugin.swift [hunk 1]
```diff
@@ -47,6 +47,16 @@ public class PushNotificationsPlugin: CAPPlugin {
...
     }
 
+    /**
+     * Unregister for remote notifications
+     */
+    @objc func unregister(_ call: CAPPluginCall) {
+        DispatchQueue.main.async {
+            UIApplication.shared.unregisterForRemoteNotifications()
+        }
+        call.resolve()
```
Comment (jcesarmobile) [inline push-notifications/ios/Plugin/PushNotificationsPlugin.swift:57]: The resolve should be inside the `DispatchQueue.main.async` block

Comment (markemer) [reply push-notifications/ios/Plugin/PushNotificationsPlugin.swift:57]: Sure, makes sense. Should `register`'s `resolve` also be in that block? I just copied the pattern.

Comment (jcesarmobile) [reply push-notifications/ios/Plugin/PushNotificationsPlugin.swift]: yeah, it should, good catch!

**Review by jcesarmobile — APPROVED** [review]

**Pull request merged**